CMakeLists.txt

```diff
@@ -7,6 +7,37 @@ Please create a directory and run cmake from there, passing the path to this sou
 This process created the file `CMakeCache.txt' and the directory `CMakeFiles'. Please delete them.")
 endif()
 
+if(ENABLE_MASON)
```

Comment:
👍  @TheMarex I wondered about that. It does appear to work above the `option` block, but I think it would conceptually make more sense below. To fix this I'll lift the `option` block up since this `ENABLE_MASON` block needs to come before `project` so that the C/CXX checks use the right compiler.